src/pages/Main.tsx

```diff
@@ -1,85 +1,120 @@
+  >({});
+  const { getLinkedIncidents } = useLinkIncidents();
+  const { getMultipleIncidentTicketCount } = useTicketCount();
 
-  // Add a "refresh" button @see https://support.deskpro.com/en-US/guides/developers/app-elements
   useInitialisedDeskproAppClient((client) => {
-    client.registerElement("myRefreshButton", { type: "refresh_button" });
-  });
+    client.setTitle("PagerDuty");
+
+    client.registerElement("homeButton", {
+      type: "home_button",
+    });
+
+    client.deregisterElement("menuButton");
+
+    client.registerElement("menuButton", {
+      type: "menu",
+      items: [
+        {
+          title: "Go to Link/Create Incident",
+          payload: {
+            type: "changePage",
+            page: "/",
+          },
+        },
+      ],
+    });
+
+    client.deregisterElement("editButton");
+
+    client.registerElement("refreshButton", {
+      type: "refresh_button",
+    });
+  }, []);
 
-  // Listen for the "change" event and store the context data
-  // as local state @see https://support.deskpro.com/en-US/guides/developers/app-events
   useDeskproAppEvents({
-    onChange: setTicketContext,
+    async onElementEvent(id) {
+      switch (id) {
+        case "menuButton":
+          navigate("/findOrCreate");
+          break;
+
+        case "homeButton":
+          navigate("/redirect");
+      }
+    },
   });
+  const incidentsByIdQuery = useQueryWithClient(
+    "getIncidentsById",
+    (client) => getIncidentsById(client, incidentIds),
+    {
+      enabled: !!incidentIds.length,
+    }
+  );
 
-  // Use the apps proxy to fetch data from a third party
-  // API @see https://support.deskpro.com/en-US/guides/developers/app-proxy
-  useInitialisedDeskproAppClient((client) =>
+  useEffect(() => {
     (async () => {
-      const fetch = await proxyFetch(client);
+      if (!context || !client) return;
 
-      const response = await fetch(
-        "https://jsonplaceholder.typicode.com/posts"
-      );
+      const linkedIncidents = await getLinkedIncidents();
 
-      const posts = await response.json();
+      if (!linkedIncidents || linkedIncidents.length === 0) {
+        navigate("/findOrCreate");
 
-      setExamplePosts(posts.slice(0, 3));
-    })()
-  );
+        return;
+      }
 
-  useInitialisedDeskproAppClient((client) => {
-    // If this is a "global" target app, then we can set the width of the app using CSS units
-    client.setWidth("50vw");
-  });
+      setIncidentIds(linkedIncidents as string[]);
+
+      const incidentLinkedCount = await getMultipleIncidentTicketCount(
+        linkedIncidents
+      );
+
+      setIncidentLinkedCount(incidentLinkedCount);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [context, client]);
+
+  const incidents = incidentsByIdQuery.data;
 
-  // If we don't have a ticket context yet, show a loading spinner
-  if (ticketContext === null) {
+  if (incidentsByIdQuery.isFetching || !incidents || !incidentLinketCount)
```

Comment:
if nothing is found, the loader spinning. How user understand that nothing was found or that the download is still in progress?

Reply:
Please handle the case, when nothing is found, then display the block with the text that nothing is found